### PR TITLE
Make Anchor Output Sweeps configurable.

### LINKED ISSUE #7602: [feature]: Make the Anchor Sweep Configurable (after commitment is confirmed)
Currently we have 2 situations where we are going to use the anchor output.

1. When the Commitment is not confirmed and either the sweeper uses the anchor to bump our commitment tx, or we use the `bumpclosefee` rpc which lets us bump the fee of a unconfirmed commitment transaction manually. This feature request does not touch these two scenarios

2. When the Commitment Tx is confirmed our anchor_resolver will try to sweep our anchor with the relayfee (253 sats/kweight). This is what this feature requests targets. Sweeping the Anchor only is economical only up to ~1.8 sat/vbyte (when sweeping before 16blocks). I propose to make this sweeping of anchors configurable. Allowing noderunners to switch off the anchor sweep either by just disabling this behaviour or maybe implement a check which  evaluates whether the fee-estimation for maybe 144 blocks would give us a higher feerate than maybe 1.5 sat/vbyte and then not sweeping the anchor programmatically.
Problem with this sweeping is, that it blocks useable outputs for the user. In bitcoin transactions can only be replaced so they would have to either replace this transaction (RBF) which makes it immediately not economical (size for p2tr output is around 194 byte), using CPFP to use the bigger output which was used to sweep this anchor makes it also not worth it, as soon as the effective feerate goes beyond 1,8 sat/vbyte. So allowing the user to switch off automatic sweeping makes totally sense in crowded mempool conditions.

I would like to tackle this issue if there is consensus that it is worth going in this direction :)

### PR DESCRIPTION
**Pull request by ziggie1984**
The sweeping of anchor outputs is not really aligned with incentives when mempool conditions do not allow 1 sat/vbyte transactions to be confirmed in a reasonable amount of time. Always sweeping anchor outputs although they have no real chance to be confirmed will cost more eventually when one needs to use the funds which were used to sweep this output. In addition anchor sweeps are mostly bumped by third parties which can lead to a mandatory restart because lnd does not monitor the mempool when RBFs happen. 
Therefore noderunners have now the possibility to switch off the anchor recovery. I did not switch it off by default because a lot of unit test would fail as a consequence.

--sweeper.notrecoveranchoroutputs=true

fixes #7602 

Checking for the tests to pass ....

### CODE REVIEW

File: contractcourt/channel_arbitrator.go [hunk 1]
```diff
@@ -771,8 +773,6 @@ func (c *ChannelArbitrator) relaunchResolvers(commitSet *CommitSet,
 		anchorResolver.SupplementState(chanState)
 
 		unresolvedContracts = append(unresolvedContracts, anchorResolver)
-
-		// TODO(roasbeef): this isn't re-launched?
```
Comment (ziggie1984) [inline contractcourt/channel_arbitrator.go:775]: not sure what you meant by this todo, according to the code the resolvers are relaunched ? The linter would complain about this comment so I removed it, happy to add it back if it still holds true.

Comment (Roasbeef) [reply contractcourt/channel_arbitrator.go:775]: IIUC, if all the contract for a resolution are resolved, but the anchor isn't, then it's never re-created.

**Comment by Roasbeef** [comment]
I don't think we want to _never_ sweep anchors. Never sweeping them means we can never use it to automatically fee bump. In that case, then `BumpFee` will need to be used to bump the fee for the anchor output. With the way things are set up right now, you still need to offer the inputs to the sweeper, as otherwise it won't know how to sweep it (it's a "special" output). 

I think something like this could be a middleground: https://github.com/lightningnetwork/lnd/pull/7965#issuecomment-1718627997

**Comment by ziggie1984** [comment]
This change would still need one or two integration tests to finalize, if we wanna go this route.

**Review by yyforyongyu — COMMENTED** [review]
I think if we want to go with this config, we need to provide a mechanism to sweep them back when requested - the easiest way is to provide a new RPC to do this, but then we'll need to persist the anchor sweep on disk, still need more thinking in this area.

Tho we know there are anchor collectors out there, I don't think we should rely on them to keep the utxo set from bloating.

**Comment by ziggie1984** [comment]
Further to this change, it should be granted that any zero-conf wallet input which is the result of a `sweep-transaction` should not be allowed to open a channel with (when using zero-conf utxos) because those can always be RBFed by the sweeper in the background leading to all sorts of side-effects.

**Comment by ziggie1984** [comment]
Obsolete with the new sweeper subsystem.

**Pull request closed without merging**